### PR TITLE
Move `make_psf`, `make_mean_psf` and `make_mean_edisp` to separate file

### PR DESCRIPTION
**Pull request by dcfidalgo**
As discussed today during the Coding Sprint, we want to move the IRF reduction away from the observation classes.
I moved them to a new file in `irf/irf_reduce.py`. For an easier review process, in this PR i just created the new file and copy/pasted the methods to it. As soon as this PR gets in, i will make a follow-up PR in which i change the methods to work as stand-alone functions.

### CODE REVIEW

**Comment by adonath** [comment]
Thanks, @dcfidalgo! As the change is rather straight forward, there is not much to review here. If you prefer, I can merge this PR and you continue in a separate one. But it's also fine by me, if you just continue your work in this PR.

**Comment by dcfidalgo** [comment]
Could you maybe just merge this one? I would prefer to open up a new PR afterwards, then you get a nicer diff for the review :)

File: gammapy/irf/irf_reduce.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import absolute_import, division, print_function, unicode_literals
+from . import PSF3D, EnergyDependentTablePSF, IRFStacker
+from ..utils.energy import Energy
+
+__all__ = ["make_psf", "make_mean_psf", "make_mean_edisp"]
+
+
+def make_psf(self, position, energy=None, rad=None):
```
Comment (cdeil) [inline gammapy/irf/irf_reduce.py]: Please rename `self` to `obs` here, and to add it to the `Parameters` list in the docstring.

`self` is  just for methods on classes, shouldn't be used for standalone functions.

Same change for the functions below, just in the last one call it `observations`.
(well, that's my preference, that we go for `observations` everywhere, currently we have a mix of `observations` and `obs_list` in Gammapy).

Comment (cdeil) [reply gammapy/irf/irf_reduce.py]: Ah, sorry, I missed your comment that you want to do that in a follow-up PR.
Yes, good, of course, remove the methods on the obs classes and use these ones.

**Pull request merged**